diff --git a/asio/include/asio/detail/impl/posix_event.ipp b/asio/include/asio/detail/impl/posix_event.ipp
index 81bea13d..3adf6564 100644
--- a/asio/include/asio/detail/impl/posix_event.ipp
+++ b/asio/include/asio/detail/impl/posix_event.ipp
@@ -31,7 +31,8 @@ namespace detail {
 posix_event::posix_event()
   : state_(0)
 {
-#if (defined(__MACH__) && defined(__APPLE__)) \
+#if defined(__MVS__) \
+      || (defined(__MACH__) && defined(__APPLE__)) \
       || (defined(__ANDROID__) && (__ANDROID_API__ < 21))
   int error = ::pthread_cond_init(&cond_, 0);
 #else // (defined(__MACH__) && defined(__APPLE__))
