diff --git a/asio/include/asio/signal_set_base.hpp b/asio/include/asio/signal_set_base.hpp
index 8c50bf58..ad94ebe5 100644
--- a/asio/include/asio/signal_set_base.hpp
+++ b/asio/include/asio/signal_set_base.hpp
@@ -62,9 +62,18 @@ public:
   enum class flags : int
   {
     none = 0,
+#ifndef __MVS__
     restart = ASIO_OS_DEF(SA_RESTART),
     no_child_stop = ASIO_OS_DEF(SA_NOCLDSTOP),
     no_child_wait = ASIO_OS_DEF(SA_NOCLDWAIT),
+#else
+    // z/OS's SA_NOCLDSTOP is 0x80000000 that can't be contained
+    // in an int, so shift-right 8 bits, it's OK because it's just
+    // an enum
+    restart = ASIO_OS_DEF(SA_RESTART) >> 8,
+    no_child_stop = ASIO_OS_DEF(SA_NOCLDSTOP) >> 8,
+    no_child_wait = ASIO_OS_DEF(SA_NOCLDWAIT) >> 8,
+#endif
     dont_care = -1
   };
 
