diff --git a/asio/include/asio/ip/unicast.hpp b/asio/include/asio/ip/unicast.hpp
index 884c6749..27dd4783 100644
--- a/asio/include/asio/ip/unicast.hpp
+++ b/asio/include/asio/ip/unicast.hpp
@@ -21,6 +21,11 @@
 
 #include "asio/detail/push_options.hpp"
 
+#ifdef __MVS__
+// need this in order to have IP_TTL (for z/OS 2.5 and below)
+#include <xti.h>
+#endif
+
 namespace asio {
 namespace ip {
 namespace unicast {
